src/main/java/com/hartwig/platinum/ApiRerun.java

```diff
@@ -28,47 +27,50 @@ public class ApiRerun {
     private final String bucket;
     private final String version;
 
-    public ApiRerun(final RunApi runApi, final SetApi setApi, final SampleApi sampleApi, final String bucket, final String version) {
+    public ApiRerun(final RunApi runApi, final SetApi setApi, final SampleApi sampleApi, final String bucket,
+            final String version) {
         this.runApi = runApi;
         this.setApi = setApi;
         this.sampleApi = sampleApi;
         this.bucket = bucket;
         this.version = version;
     }
 
-    Long create(final String sampleId) {
-        try {
-            List<Sample> samples = sampleApi.list(null, null, null, null, SampleType.TUMOR, sampleId);
-            for (Sample sample : samples) {
-                Optional<SampleSet> maybeSampleSet = OnlyOne.ofNullable(setApi.list(null, sample.getId(), true), SampleSet.class);
-                if (maybeSampleSet.isPresent()) {
-                    SampleSet sampleSet = maybeSampleSet.get();
-                    if (runApi.list(null, null, sampleSet.getId(), null, null, null, null, null)
-                            .stream()
-                            .anyMatch(r1 -> r1.getStatus().equals(Status.VALIDATED))) {
-                        return OnlyOne.ofNullable(runApi.list(null, Ini.RERUN_INI, sampleSet.getId(), version, version, null, null, null),
-                                Run.class).filter(r -> !r.getStatus().equals(Status.INVALIDATED)).map(r -> {
-                            LOGGER.info("Using existing run for sample [{}] id [{}]", sampleId, r.getId());
-                            return r.getId();
-                        }).orElseGet(() -> {
-                            final Long id = runApi.create(new CreateRun().bucket(bucket)
-                                    .cluster("gcp")
-                                    .context("RESEARCH")
-                                    .ini(Ini.RERUN_INI)
-                                    .version(version)
-                                    .status(Status.PENDING)
-                                    .isHidden(true)
-                                    .setId(sampleSet.getId())).getId();
-                            LOGGER.info("Created API run for sample [{}] id [{}]", sampleId, id);
-                            return id;
-                        });
-                    }
-                }
-            }
-        } catch (Exception e) {
-            LOGGER.error("Unable to create run for [{}] reason [{}]", sampleId, e.getMessage());
-            return null;
+    public Long create(final String tumorSampleName) {
```

Comment:
Simplified this a bit with removing the check for validates runs. That input validation should be done upstream, the user should decide whether a sample is to be rerun, and not platinum